mlir/lib/Conversion/MIOpenToGPU/MIOpenToGPU.cpp

```diff
@@ -123,6 +112,22 @@ void LowerMIOpenOpsToGPUPass::runOnOperation() {
     symbolTable.insert(theGpuModule);
   }
 
+  // Check parameters and populate default values if necessary.
+  SmallVector<StringRef, 1> kernelNameTable;
+  if (kernelNameList.empty()) {
+    for (auto func : op.getOps<FuncOp>())
+      if (func->getAttr("kernel"))
+        kernelNameTable.push_back(func.getName());
+  } else {
+    // Split kernelNameList into a vector separated with comma.
+    StringRef remainingKernelNameList = kernelNameList;
+    do {
+      std::pair<StringRef, StringRef> p = remainingKernelNameList.split(',');
+      kernelNameTable.push_back(p.first);
+      remainingKernelNameList = p.second;
+    } while (!remainingKernelNameList.empty());
+  }
```

Comment:
Now it becomes more clear to me:
 - If client set a kernelNameList, use the client options (Like the existing API option)
 - Else, populate the default (instead a single name, populate a list using name of funcOp)